### ISSUE TITLE
List snapshots with multiple filters tag and host filter not working ?

### ISSUE THREAD

**Issue by SkYNewZ** [restic/restic#2688]
Output of `restic version`
-----------------
`restic 0.9.6 compiled with go1.13.4 on darwin/amd64`

How did you run restic exactly?
-----------------
```
~ ➜ restic snapshots                                       

repository 1a5af6f7 opened successfully, password is correct
ID        Time                 Host                   Tags              Paths
-----------------------------------------------------------------------------------------------------
313fa2a7  2020-04-11 14:40:22  kimsufi                minecraft,simon   /data/minecraft-server/simon
cf0b237f  2020-04-11 14:40:37  kimsufi                minecraft,survie  /data/minecraft-server/survie
8260b61a  2020-04-11 14:49:02  kimsufi                minecraft,simon   /data/minecraft-server/simon
c4db6d76  2020-04-11 14:49:18  kimsufi                minecraft,survie  /data/minecraft-server/survie
d12127dd  2020-04-11 14:55:38  kimsufi                minecraft,simon   /data/minecraft-server/simon
59104702  2020-04-11 14:55:54  kimsufi                minecraft,survie  /data/minecraft-server/survie
039abf77  2020-04-11 15:05:51  kimsufi                minecraft,simon   /data/minecraft-server/simon
4b6f9eb0  2020-04-11 15:06:06  kimsufi                minecraft,survie  /data/minecraft-server/survie
980e3bbf  2020-04-11 15:27:00  kimsufi                rancher           /rancher-database.sql
8bbdfd9a  2020-04-11 15:29:00  kimsufi                rancher           /rancher-database.sql
97ec3fab  2020-04-11 16:24:49  kimsufi                minecraft,simon   /data/minecraft-server/simon
0cfe2915  2020-04-11 16:25:06  kimsufi                minecraft,survie  /data/minecraft-server/survie
c18cae9e  2020-04-11 16:34:19  management.aloesia.fr  aloesia           /aloesia-database.sql
c2fa782d  2020-04-11 16:37:40  kimsufi                minecraft,simon   /data/minecraft-server/simon
0182cd9f  2020-04-11 16:37:56  kimsufi                minecraft,survie  /data/minecraft-server/survie
0bd84c34  2020-04-11 16:38:13  kimsufi                minecraft,config  /server.properties
-----------------------------------------------------------------------------------------------------
16 snapshots
~ ➜ restic snapshots --host kimsufi --tag "minecraft,simon"
repository 1a5af6f7 opened successfully, password is correct
```

What backend/server/service did you use to store the repository?
-----------------
`GCS`

Expected behavior
-----------------

```
~ ➜ restic snapshots --host kimsufi --tag "minecraft,simon"
repository 1a5af6f7 opened successfully, password is correct

ID        Time                 Host                   Tags              Paths
-----------------------------------------------------------------------------------------------------
313fa2a7  2020-04-11 14:40:22  kimsufi                minecraft,simon   /data/minecraft-server/simon
8260b61a  2020-04-11 14:49:02  kimsufi                minecraft,simon   /data/minecraft-server/simon
d12127dd  2020-04-11 14:55:38  kimsufi                minecraft,simon   /data/minecraft-server/simon
039abf77  2020-04-11 15:05:51  kimsufi                minecraft,simon   /data/minecraft-server/simon
97ec3fab  2020-04-11 16:24:49  kimsufi                minecraft,simon   /data/minecraft-server/simon
c2fa782d  2020-04-11 16:37:40  kimsufi                minecraft,simon   /data/minecraft-server/simon
-----------------------------------------------------------------------------------------------------
6 snapshots
```

Actual behavior
---------------

Not snapshots found, but no errors neither

Steps to reproduce the behavior
-------------------------------

1. Backup with a tag
2. List snapshots filtering by host and this tag

Do you have any idea what may have caused this?
-----------------------------------------------

Maybe is not supported ? But there is no errors

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

Sure !!

**Comment by SkYNewZ** [comment]
With another tag on another host, its seem to work
```
~ ➜ restic snapshots --host management.aloesia.fr --tag aloesia
repository 1a5af6f7 opened successfully, password is correct
ID        Time                 Host                   Tags        Paths
---------------------------------------------------------------------------------------
c18cae9e  2020-04-11 16:34:19  management.aloesia.fr  aloesia     /aloesia-database.sql
---------------------------------------------------------------------------------------
1 snapshots
```

**Comment by dimejo** [comment]
That's strange. I just tried it out and it's working for me as expected. I can't spot any typos or similiar on your side but what's different between your 1st and 2nd approach are the double quotes on the tags. It shouldn't make a difference but can you try again without quotes?

**Comment by SkYNewZ** [comment]
Same issue…
```
~ ➜ restic snapshots --host kimsufi --tag "minecraft,simon"
repository 1a5af6f7 opened successfully, password is correct

~ ➜ restic snapshots --host kimsufi --tag minecraft,simon  
repository 1a5af6f7 opened successfully, password is correct

~ ➜ restic snapshots --host kimsufi --tag minecraft --tag simon
repository 1a5af6f7 opened successfully, password is correct
```

Even a single tag does not work for this host 🤔
```
~ ➜ restic snapshots --host kimsufi --tag minecraft            
repository 1a5af6f7 opened successfully, password is correct

```

**Comment by MichaelEischer** [comment]
Could you post the tag-list as printed by `restic cat snapshot 313fa2a7` (or one of the other snapshots with the "minecraft,simon" tag). The backup command is not splitting tags on commas, but it looks like the snapshot command does. My guess is that your snapshots have a "minecraft,simon" tag and not "minecraft" and "simon".

**Comment by SkYNewZ** [comment]
You are right…
```
~ ➜ restic cat snapshot 313fa2a7                              
repository 1a5af6f7 opened successfully, password is correct
{
  "time": "2020-04-11T14:40:22.014491396+02:00",
  "tree": "530883dffc628f8cff7c04f2d8e7ab1863f5f36c58faadea6f22df3754b6356d",
  "paths": [
    "/data/minecraft-server/simon"
  ],
  "hostname": "kimsufi",
  "username": "quentin",
  "uid": 1000,
  "gid": 1000,
  "tags": [
    "minecraft,simon"
  ]
}
```

I'm using this to backup:
```
${RESTIC_BIN} backup -vvv "${MAP_ROOT_PATH}/${map}" --tag "minecraft,${map}"
```

This can explain why doing `restic forget --prune --keep-last 2 --tag minecraft,survie` does nothing 😅

**Comment by SkYNewZ** [comment]
As expected, change to `${RESTIC_BIN} backup -vvv "${MAP_ROOT_PATH}/${map}" --tag minecraft --tag ${map} …` fixed my issue ! Thanks for your help

**Comment by SkYNewZ** [comment]
- `forget` have `--tag taglist            only consider snapshots which include this taglist in the format 'tag[,tag,…]'`
- `backup` have `--tag taglist       only consider snapshots which include this taglist (can be specified multiple times) (default [])`
- `snapshots` have same as `backup`

**Comment by SkYNewZ** [comment]
I've been trying to investigate on my own to try to help you. Unfortunately without success. At the risk of not teaching you anything, I just discovered that the entry-points of these 3 commands got the tags as a list, even using commas as separator…

<img width="634" alt="image" src="https://user-images.githubusercontent.com/26750012/79085223-f4165900-7d37-11ea-95dd-3bd555e2acec.png">
<img width="489" alt="image" src="https://user-images.githubusercontent.com/26750012/79085232-01334800-7d38-11ea-849f-c5e5d75c86ce.png">

**Comment by ghstwhl** [comment]
I was going to open a new issue report tonight, but I think my issue is closely related to this issue.  If you think otherwise, let me know and I'll open a new issue.

I have a repo that I have backed up into from multiple hosts.  I would like to prune all but the last X backup from each of the hosts that have written to the repo.  The [documentation](https://github.com/restic/restic/blob/master/doc/060_forget.rst) seemed to indicate that I could stack policies and they would be ORed together.  There is an explicit example of doing this with --tag parameters.  When I tried this with multiple --host parameters, only the last one on the command line is honored and the preceding ones are ignored.

Is this expected behavior?
